check.py

```diff
@@ -413,6 +413,7 @@ def fix(x):
     print s
     f = open('a.js', 'w')
     f.write(open(os.path.join(options.binaryen_bin, 'binaryen.js')).read())
+    f.write('var Binaryen = module.exports;\n')
```

Comment:
The way it works now is:

1. [Check if binaryen.js is run in a CommonJS context](https://github.com/dcodeIO/binaryen/blob/4f62b7fd88a4f59338af2b26dd46cf55c6a909f7/src/js/binaryen.js-post.js#L914). If so, export `Module`
2. [Check if binaryen.js is run in an AMD context](https://github.com/dcodeIO/binaryen/blob/4f62b7fd88a4f59338af2b26dd46cf55c6a909f7/src/js/binaryen.js-post.js#L916). If so, define a factory method that returns `Module`
3. [Otherwise, expose Binaryen as a global var](https://github.com/dcodeIO/binaryen/blob/4f62b7fd88a4f59338af2b26dd46cf55c6a909f7/src/js/binaryen.js-post.js#L918)

The reason why the additional line is necessary in check.py is that the generated a.js is running inside of a CommonJS context, but not used like that because the tests are just concatenated. That's merely a special case of the test suite.